integrations/micronaut/cdi/src/main/java/io/helidon/integrations/micronaut/cdi/MicronautMethodInvocationContext.java

```diff
@@ -0,0 +1,221 @@
+
+    private final InvocationContext cdiContext;
+    private final ExecutableMethod executableMethod;
+    private final Set<MethodInterceptor<?, ?>> allInterceptors;
+    private final MutableConvertibleValues attributes;
+    private final Map<String, MutableArgumentValue<?>> mutableArguments;
+
+    private Iterator<MethodInterceptor<?, ?>> remaining;
+
+    private MicronautMethodInvocationContext(InvocationContext cdiContext,
+                                             ExecutableMethod executableMethod,
+                                             Set<MethodInterceptor<?, ?>> allInterceptors,
+                                             Iterator<MethodInterceptor<?, ?>> remaining) {
+        this.cdiContext = cdiContext;
+        this.executableMethod = executableMethod;
+        this.allInterceptors = allInterceptors;
+        this.remaining = remaining;
+
+        this.attributes = new MutableConvertibleMultiValuesMap();
+        this.mutableArguments = new LinkedHashMap<>();
+        Object[] parameters = cdiContext.getParameters();
+        Argument[] argumentNames = executableMethod.getArguments();
+
+        for (int i = 0; i < argumentNames.length; i++) {
+            Argument argument = argumentNames[i];
+            Object parameterValue = parameters[i];
+            mutableArguments.put(argument.getName(),
+                                 new MutableArgument<Object>(argument, parameterValue));
+        }
+    }
+
+    static MethodInvocationContext create(InvocationContext cdiCtx,
+                                          ExecutableMethod<?, ?> executableMethod,
+                                          Set<MethodInterceptor<?, ?>> allInterceptors,
+                                          Iterator<MethodInterceptor<?, ?>> remaining) {
+        return new MicronautMethodInvocationContext(cdiCtx,
+                                                    executableMethod,
+                                                    allInterceptors,
+                                                    remaining);
+    }
+
+    @Override
+    public ExecutableMethod getExecutableMethod() {
+        return executableMethod;
+    }
+
+    @Override
+    public Map<String, MutableArgumentValue<?>> getParameters() {
+        return mutableArguments;
+    }
+
+    @Override
+    public Object getTarget() {
+        return cdiContext.getTarget();
+    }
+
+    @Override
+    public Object proceed() throws RuntimeException {
+        if (remaining.hasNext()) {
+            MethodInterceptor<?, ?> next = remaining.next();
+            LOGGER.finest(() -> "Micronaut interceptor: " + next.getClass().getName());
+            return next.intercept(this);
+        }
+        try {
+            Object[] arguments = mutableArguments.values()
+                    .stream()
+                    .map(ArgumentValue::getValue)
+                    .toArray(Object[]::new);
```

Comment:
Again in Micronaut core we tend to have an implementation whereby `MutableArgumentValue.setValue(..)` will set the value into the `Object[]` to avoid this copying